src/org/zalando/stups/mint/worker/external/bucket_storage.clj

```diff
@@ -0,0 +1,14 @@
+(ns org.zalando.stups.mint.worker.external.bucket_storage)
+
+(defmacro StorageException
+  [msg data]
+  `(ex-info ~msg (merge ~data {:type "StorageException"})))
+
+(defn storage-exception? [e]
+  (= "StorageException" (:type (ex-data e))))
+
+(defprotocol BucketStorage
```

Comment:
Thanks, I'm not really well versed in clojure (yet) as you might have noticed, but this looks like a good way to do it. I will look into the defmulti approach.